src/SlashCommands.tsx

```diff
@@ -220,12 +220,16 @@ const isCurrentLocalRoom = (): boolean => {
 export const Commands = [
     new Command({
         command: 'spoiler',
-        args: '<message>',
+        args: '[!<warning>!] <message>',
         description: _td('Sends the given message as a spoiler'),
         runFn: function(roomId, message) {
+            const regex = /!([^!]+)!/;
+            const result = regex.exec(message);
+            const attr = result == null ? "" : `="${result[1]}"`;
+
             return successSync(ContentHelpers.makeHtmlMessage(
                 message,
-                `<span data-mx-spoiler>${message}</span>`,
+                `<span data-mx-spoiler${attr}>${message.replace(regex, "")}</span>`,
```

Comment:
The formatting will easily break here if the spoiler warning contains any `"` characters, or if the message contains a `</span>`. To fix this, you could create a span element [in a new document](https://developer.mozilla.org/en-US/docs/Web/API/DOMImplementation/createHTMLDocument) and then ask the browser to produce the escaped HTML for you